catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/MainMenuFragment.java

```diff
@@ -157,10 +164,21 @@ public void onItemClick(@ButtonId int id) {
 				startActivity(new Intent(getActivity(), WebViewActivity.class));
 				break;
 			case UPLOAD:
-				setShowProgressBar(true);
-				Intent intent = new Intent(getActivity(), ProjectUploadActivity.class)
-						.putExtra(ProjectUploadActivity.PROJECT_NAME, Utils.getCurrentProjectName(getActivity()));
-				startActivity(intent);
+				Project currentProject = null;
+				try {
+					currentProject = XstreamSerializer.getInstance().loadProject(Utils.getCurrentProjectName(getActivity()), getActivity());
+				} catch (IOException | LoadingProjectException e) {
+					Log.e(TAG, e.getMessage());
+				}
+				if (Utils.isDefaultProject(currentProject, getActivity())) {
```

Comment:
here the `currentProject` is still `null` if an exception occurred -> it might be best to just move this code into the `try` block.